schemas/typescript/types.ts

```diff
@@ -107,148 +204,395 @@ export type QuotePreference =
+ * @example
+ * {
+ *   type: "oif-3009-v0",
+ *   payload: {
+ *     signatureType: "eip712",
+ *     domain: { name: "USD Coin", version: "2", chainId: 1, verifyingContract: "0xA0b8..." },
+ *     primaryType: "TransferWithAuthorization",
+ *     message: {
+ *       from: "0x000100000101742d35Cc6634C0532925a3b844Bc9e7595f0bEb8",
+ *       to: "0x00010000010195ad61b0a150d79219dcf64e1e6cc01f0c0c8a4a",
+ *       value: "1000000000",
+ *       validAfter: 0,
+ *       validBefore: 1700000000,
+ *       nonce: "0xabcd1234..."
+ *     }
+ *   },
+ *   metadata: {
+ *     orderHash: "0xdef456...",
+ *     chainId: 1,
+ *     tokenAddress: "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48"
+ *   }
+ * }
  */
-export interface Eip712Order {
-  /** Domain for the EIP-712 order */
-  domain: Address;
-  /** Primary type of the EIP-712 message */
-  primaryType: string;
-  /** EIP-712 message content */
-  message: Record<string, unknown>;
+export interface Oif3009Order {
+  /** Order type identifier for EIP-3009 transfers */
+  type: "oif-3009-v0";
+  /** EIP-3009 Transfer With Authorization typed data */
+  payload: {
+    /** Signature type indicator */
+    signatureType: "eip712";
+    /** EIP-712 domain separator */
+    domain: object;
+    /** Primary type name (typically "TransferWithAuthorization") */
+    primaryType: string;
+    /** The transfer authorization message */
+    message: object;
+  };
+  /** Additional metadata for nonce verification and order tracking */
+  metadata: object;
 }
 
 /**
- * Detailed information about a quote
- * @description Detailed information about a quote
+ * Generic order type
+ * @description Flexible order format that can accommodate various signature schemes and
+ *              order structures. Expects a similar structure to other order types with
+ *              signatureType and associated EIP-712 or other signature data.
+ * @example With EIP-712:
+ * {
+ *   type: "oif-generic-v0",
+ *   payload: {
+ *     signatureType: "eip712",
+ *     domain: { name: "Custom Protocol", version: "1", chainId: 1, verifyingContract: "0x..." },
+ *     primaryType: "LimitOrder",
+ *     message: {
+ *       maker: "0x000100000101742d35Cc6634C0532925a3b844Bc9e7595f0bEb8",
+ *       inputToken: "0x000100000101A0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
+ *       outputToken: "0x000100000101dAC17F958D2ee523a2206206994597C13D831ec7",
+ *       inputAmount: "1000000000",
+ *       outputAmount: "999000000",
+ *       expiry: 1700000000
+ *     }
+ *   }
+ * }
  */
-export interface QuoteDetails {
-  /** Requested outputs in the quote */
-  requestedOutputs: RequestedOutputDetails[];
-  /** Available inputs in the quote */
-  availableInputs: Array<{
-    /** User address */
-    user: Address;
-    /** Asset address */
-    asset: Address;
-    /** Amount available */
-    amount: Amount;
-    /** Lock type - if empty, the asset needs to be escrowed */
-    lockType?: "the-compact";
-  }>;
+export interface OifGenericOrder {
```

Comment:
What's a generic order? I think this was an idea that should've been removed, wasn't it?